File: Day1/day1.py
import sys

if __name__ == "__main__":
    try:
        file_name = sys.argv[1]
    except IndexError:
        file_name = "input.txt"      
    text = open(file_name).read().split(sep = "\n\n")
    text = [[int(item) for item in sublist.split("\n") if item != ""] for sublist in text]
    calories = sorted(map(sum, text), reverse = True)

    print("Top: ", calories[0])
    print("Top 3: ", sum(calories[:3]))

File: Day1/day1.rs
fn get_data() -> String {
    let input_args: Vec<String> = std::env::args().collect();
    let mut file_name: String = "input.txt".to_string();
    if input_args.len() > 1 {
        file_name = input_args[1].clone();
    }
    let mut data = std::fs::read_to_string(file_name).unwrap();
    if data.ends_with("\n") {
        data.pop();
    }
    return data;
}

fn main() {
    let data = get_data();
    let mut calories: Vec<_> = data
        .split("\n\n")
        .map(|element| {
            element
                .lines()
                .map(|item| item.parse::<u32>().unwrap())
                .sum::<u32>()
        })
        .collect();
    calories.sort();
    calories.reverse();
    println!("Top: {}", calories[0]);
    let biggest_three = calories[0..3].iter().sum::<u32>();
    println!("Top three: {}", biggest_three)
}
